memory/memhotplug.py

```diff
@@ -0,0 +1,250 @@
+        for block in blocks:
+            if not memory._check_memory_state(block):
+                online(block)
+
+    @staticmethod
+    def __clear_dmesg():
+        process.run("dmesg -c", sudo=True)
+
+    @staticmethod
+    def __error_check():
+        ERROR = None
+        logs = (process.system_output(
+            "dmesg -T --level=alert,crit,err,warn", ignore_status=True, shell=True, sudo=True)).split()
+        for log in logs:
+            for error in errorlog:
+                if error in log:
+                    ERROR = 'Test: resulted %s in syslogs"' % error
+        return ERROR
+
+    @staticmethod
+    def __is_auto_online():
+        with open('%s/auto_online_blocks' % mem_path, 'r') as auto_file:
+            if auto_file.read() == 'online\n':
+                return True
+            return False
+
+    @staticmethod
+    def run_stress(stresstime):
+        mem_free = memory.meminfo.MemFree.mb / 2
+        cpu_count = int(multiprocessing.cpu_count()) / 2
+        process.run("stress --cpu %s --io 4 --vm 2 --vm-bytes %sM --timeout %ss" %
+                    (cpu_count, mem_free, stresstime), ignore_status=True, sudo=True, shell=True)
+
+    def test(self):
+        if os.path.exists("%s/auto_online_blocks" % mem_path):
+            if not self.__is_auto_online():
+                self.hotplug_all(self.blocks_hotpluggable)
+        if 'all' in self.tests:
+            tests = ['hotplug_loop',
+                     'hotplug_toggle',
+                     'hotplug_ratio',
+                     'dlpar_mem_hotplug',
+                     'hotplug_per_numa_node']
+        else:
+            tests = self.tests.split()
+
+        for method in tests:
+            self.log.info("\nTEST: %s\n", method)
+            self.__clear_dmesg()
+            run_test = 'self.%s()' % method
+            eval(run_test)
+            msg = self.__error_check()
+            if msg:
+                collect_dmesg()
+                self.log.error('Test: %s. ERROR Message: %s', run_test, msg)
+            self.log.info("\nEND: %s\n", method)
+
+    def hotplug_loop(self):
+        self.log.info("\nTEST: hotunplug and hotplug in a loop\n")
+        for _ in range(self.iteration):
+            self.log.info("\nhotunplug all memory\n")
+            self.hotunplug_all(self.blocks_hotpluggable)
+            self.run_stress(self.stresstime)
+            self.log.info("\nReclaim back memory\n")
+            self.hotplug_all(self.blocks_hotpluggable)
+
+    def hotplug_toggle(self):
+        self.log.info("\nTEST: Memory toggle\n")
+        for _ in range(self.iteration):
+            for block in self.blocks_hotpluggable:
+                offline(block)
+                self.log.info("memory%s block hotunplugged" % block)
+                self.run_stress(self.stresstime)
+                online(block)
+                self.log.info("memory%s block hotplugged" % block)
+
+    def hotplug_ratio(self):
+        for ratio in ('25', '50', '75', '100'):
```

Comment:
is this ratio is random just want to understand can you please help me in same , if yes than i think it can also be as a yaml input

Reply:
ratio is random, it also goof idea to pass from yaml